crates/sui-framework/sources/crypto/bls12381.move

```diff
@@ -25,8 +29,17 @@ module sui::bls12381 {
         msg: vector<u8>,
         domain: vector<u8>
     ): bool {
-        std::vector::append(&mut domain, msg);
-        bls12381_min_sig_verify(signature, public_key, &domain)
+        // The first 3 bytes represents the intent of the message, corresponding to 
+        // IntentScope::ProofOfPossession = 5, IntentVersion::V0 = 0, AppId::Sui = 0. 
+        // The next 2 bytes corresponds to BCS serialization perfix of an IntentMessage<T>. 
+        let signed_bytes = vector[5, 0, 0, 132, 1];
+        // Append domain. 
+        std::vector::append(&mut signed_bytes, domain);
+        // Append the message.
+        std::vector::append(&mut signed_bytes, msg);
+        // Append the BCS bytes of the epoch_id: u64 = 0. 
+        std::vector::append(&mut signed_bytes, vector[0, 0, 0, 0, 0, 0, 0, 0]);
+        bls12381_min_sig_verify(signature, public_key, &signed_bytes)
```

Comment:
here we try to reconstruct the intent message from the message itself, there are some magic numbers here that I annotated heavily what this means. 

alternatively, we can change this method to use an native function, where we can match our rust impl 
```pop
        .verify_secure(
            &IntentMessage::new(
                Intent::default().with_scope(IntentScope::ProofOfPossession),
                msg
            ),
            0,
            kp.public().into()
        )
```

which way do we prefer here? @kchalkias

Reply:
I don't have much context here, but today we verify validator metadata using a [rust native function](https://github.com/MystenLabs/sui/blob/main/crates/sui-types/src/sui_system_state.rs#L106). This excludes protocol pubkey + its pop, which is yet to migrate. We could just add the pop verify part in that function. Although a new native function may be desired in the future for bls key verify

Reply:
whats the difference btw verifying this as a metadata vs here (when request to add to validator)?: https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/sources/governance/validator.move#L208

are these two different flows?

Reply:
discussed offline, removed the verify_proof_of_possession call in validator add flow, since we have it in  validate_metadata to use rust native impl. see second commit for all sui-framework changes